src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcTransport.cs

```diff
@@ -48,6 +48,88 @@ internal abstract class IpcEndpoint
         public abstract Task WaitForConnectionAsync(CancellationToken token);
     }
 
+    internal class IpcEndpointConfig
+    {
+        internal enum PortType
+        {
+            None,
+            Connect,
+            Listen
+        }
+
+        public string Address { get; } = "";
+        public PortType Type { get; } = PortType.None;
+
+        public IpcEndpointConfig(string config)
+        {
+            if (!string.IsNullOrEmpty(config))
+            {
+                var parts = config.Split(',');
+
+                Address = parts[0];
+                Type = PortType.Listen;
+
+                for (int i = 1; i < parts.Length; i++)
+                {
+                    if (string.Compare(parts[i], "connect", StringComparison.OrdinalIgnoreCase) == 0)
```

Comment:
What's the benefit, except comparing against bool instead of == 0?

Reply:
Depends on the compare implementation, but compare doesn't have to just stop at the first differing character; it could compute some value for both strings and then compare them. Whereas, equality comparing will check character-by-character and immediately stop after the first difference.

Reply:
When you do case sensitive compare (StringComparison.Ordinal),  equals can do optimized check, BCL implements it as:

```
SpanHelpers.SequenceEqual(
    ref Unsafe.As<char, byte>(ref strA.GetRawStringData()),
    ref Unsafe.As<char, byte>(ref strB.GetRawStringData()),
    ((uint)strA.Length) * sizeof(char));
```

but when you do a case insensitive compare the implementation depends on string content and in some scenarios both implementations get very close or pretty much identical, but the fact that the Equals is not required to report different results (just bool) while Compare needs to support sorting order puts Equals in position to accept more implementation optimizations.

Looking in codebases like mono/mono and .net framework reference source those tend to favour string.Compare for case insensitive compare, while dotnet/runtime seems to favour string.Equals. So seems to shifted in favour of string.Equals over time, interesting!